lib/sonar/cli/cli.rb

```diff
@@ -27,7 +29,12 @@ def usage
 
     desc 'search_all [QUERY TERM]', 'Search all Sonar record types'
     def search_all(term)
-      Sonar::Search::QUERY_TYPES_MAP.keys.each do |type|
+      if term =~ IS_IP
+        types = ip_search_type_names
+      else
+        types = domain_search_type_names
+      end
+      types.each do |type|
         search(type, term) rescue 'Search failed'
```

Comment:
Okay merged that so that problem is gone.  I believe the other empty collection issue is handled by the `format` option like so: 

```
~/work/sonar-client(branch:feature/sonar-all-command*) » sonar search namecerts marklin-users.net    
------------------------------------------------------------
~/work/sonar-client(branch:feature/sonar-all-command*) » sonar search namecerts marklin-users.net --format flat
{"collection":[],"more":false,"warning":"","iterator_id":"78c614cc2cf70a5f"}
```
Such that `flat` takes whatever is in collection and displays it.  Whereas the raw (non-flat, which I believe is called `lines`) format will show an empty collection.  The terminology is confusing but I believe has the desired effect.